packages/esm-service-queues-app/src/patient-search/hooks/useQueueLocations.tsx

```diff
@@ -10,11 +10,11 @@ interface FHIRResponse {
 }
 export function useQueueLocations() {
   const apiUrl = `${fhirBaseUrl}/Location?_summary=data&_tag=queue location`;
-  const { data, error, isLoading } = useSWR<{ data: FHIRResponse }>(apiUrl, openmrsFetch);
+  const { data, error } = useSWR<{ data: FHIRResponse }>(apiUrl, openmrsFetch);
 
   const queueLocations = useMemo(
     () => data?.data?.entry?.map((response) => response.resource) ?? [],
     [data?.data?.entry],
   );
-  return { queueLocations: queueLocations ? queueLocations : [], isLoading, error };
+  return { queueLocations: queueLocations ? queueLocations : [], isLoading: !data && !error, error };
```

Comment:
I've seen this done elsewhere in the code base, but not sure why. Any reason to not use the `isLoading` from `useSWR`?

Reply:
Yes, we can use `isLoading` directly from `useSWR`. So what I've done I made `isLoading` customized based on additional conditions `(!data && !error)`.
It implies that the component is considered loading when there is neither data nor an error.

Reply:
Hi @Vijaykv5 @chibongho !
This was actually being done earlier with useSWR v1, where we didn't use to get `isLoading` from `useSWR`, but after upgrading to useSWR v2, we don't need to handle the case for isLoading anymore.

> So what I've done I made isLoading customized based on additional conditions (!data && !error).
Actually the `isLoading` originally is defined the same. We have another returned variable from `useSWR` namely `isValidating`, which is true whenever the request is sent, which might not be the first request.

Read here for more info on isLoading and isValidating: https://swr.vercel.app/docs/advanced/understanding.en-US#combining-with-isloading-and-isvalidating-for-better-ux